python_ta/typecheck/type_store.py

```diff
@@ -88,6 +95,39 @@ def lookup_method(self, operator, *args):
                     return func_type
             raise KeyError
 
+    def node_to_type(self, node):
+        if isinstance(node, astroid.Name):
+            try:
+                inner_var = globals()[node.name]
+            except KeyError:
+                if node.name in _TYPESHED_TVARS:
+                    inner_var = _TYPESHED_TVARS[node.name]
+                else:
+                    inner_var = _ForwardRef(node.name)
+            return inner_var
+        elif isinstance(node, astroid.Subscript):
+            if hasattr(node.slice.value, 'name') and \
+                    isinstance(node.slice.value, astroid.Tuple):
+                inner_var = tuple([self.node_to_type(elt) for elt in node.slice.value.elts])
+            else:
+                inner_var = self.node_to_type(node.slice.value)
+            try:
+                outer_var = globals()[node.value.name]
+            except KeyError:
+                outer_var = _ForwardRef(node.value.name)
+            return outer_var[inner_var]
+
+    def is_descendant(self, child: type, ancestor: type):
```

Comment:
Why aren't you using `mro` here?

Reply:
For some reason mro does not return base classes like Sequence[~T] or SupportsAbs[int], that's the reason why I'm storing `__bases` separately and using this function (for regular non-mro inheritance). There still remains the issue of how to deal with method resolution that depends on classes like these.